### PR TITLE
Store details step - add contextual information and skip link

### LINKED ISSUE #4566: Store details step - add contextual information and skip link
**User story:**
As a new merchant
I want to have more context on why I should add info about my store and having a way to skip those steps
So I can decide if I should provide that info or skip it entirely 

**Description:** with the removal of the splash screen we need to adjust the header to provide a more friendly welcome, add a skip link beneath the card and informational tooltips to provide more context.

**Acceptance Criteria:**
- [x] Update the header content
- [x] Add a tooltip in the subtitle explaining why it's important to provide this info
- [x] Add a link to skip the Store Profiler
- [x] Add a tooltip in the skip link reminding why merchants should avoid skipping it

**Event tracking:**
Event name: wcadmin_storeprofiler_store_details_skip
Event prop: NA
Description: when the user clicks to skip the profiler

**Additional links:**
[Design mockup](https://www.figma.com/file/JH9XMFUCOjfXdr3N09AHRD/On-boarding-iterations-June-'20?node-id=171%3A2)

### PR DESCRIPTION
**Pull request by samueljseay**
Fixes #4566 

Changes:

* Updated `@wordpress/components` and `@wordpress/base-styles`
* Use the new `Card` and `Flex` from `@wordpress/components`
* Add contextual tooltips
* Adjust existing styles to match [new designs](https://www.figma.com/file/JH9XMFUCOjfXdr3N09AHRD/On-boarding-iterations-June-'20)

### Accessibility

- [ ] I've tested using only a keyboard (no mouse)
- [ ] I've tested using a screen reader
- [x] All animations respect [`prefers-reduced-motion`](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion)
- [x] All text has [at least a 4.5 color contrast with its background](https://webaim.org/resources/contrastchecker/)

### Screenshots
Screenshot 2020-07-02 at 3 58 35 PM
Screenshot 2020-07-02 at 3 58 48 PM

### Detailed test instructions:

- Start the setup wizard via the usual means (either after installing WooCommerce the first time or via `WooCommerce > Settings > Help > Setup Wizard`)
- Visually confirm that the page matches [the designs](https://www.figma.com/file/JH9XMFUCOjfXdr3N09AHRD/On-boarding-iterations-June-'20)
- Visually confirm that the tooltips work on hover and that the content is correct.
- You should also test that changing a user's color scheme still works. There are some good instructions on testing that [here](https://github.com/woocommerce/woocommerce-admin/pull/4558)

### CODE REVIEW

File: client/profile-wizard/steps/store-details.js [hunk 1]
```diff
@@ -26,10 +35,11 @@ import {
 } from 'dashboard/components/settings/general/store-address';
 import UsageModal from './usage-modal';
 import { CurrencyContext } from 'lib/currency-context';
+import { recordEvent } from 'lib/tracks';
 
 class StoreDetails extends Component {
 	constructor( props ) {
-		super( ...arguments );
+		super( props );
```
Comment (samueljseay) [inline client/profile-wizard/steps/store-details.js:42]: this was just a formatting change, really no need to pass anything but props to `Component`.

File: client/profile-wizard/steps/store-details.js [hunk 2]
```diff
@@ -167,67 +177,110 @@ class StoreDetails extends Component {
 		return (
 			<Fragment>
 				<H className="woocommerce-profile-wizard__header-title">
-					{ __( 'Where is your store based?', 'woocommerce-admin' ) }
+					{ __( 'Welcome to WooCommerce', 'woocommerce-admin' ) }
 				</H>
 				<H className="woocommerce-profile-wizard__header-subtitle">
 					{ __(
-						'This will help us configure your store and get you started quickly',
+						"Tell us about your store and we'll get you set up in no time",
```
Comment (samueljseay) [inline client/profile-wizard/steps/store-details.js:184]: A question: do I need to do anything for translations in other languages here?

File: client/profile-wizard/steps/store-details.js [hunk 3]
```diff
@@ -167,67 +177,110 @@ class StoreDetails extends Component {
...
-				</H>
 
-				<Card>
-					<Form
-						initialValues={ this.initialValues }
-						onSubmitCallback={ this.onSubmit }
-						validate={ validateStoreAddress }
+					<Tooltip
+						text={ __(
+							'Your store address will help us configure currency\n options and shipping rules automatically.\n This information will not be publicly visible and can\n easily be changed later.',
```
Comment (samueljseay) [inline client/profile-wizard/steps/store-details.js:190]: I used newlines and whitespace: pre to get this displaying as close to the designs as possible. This seems problematic for translations?

I tried modifying elements on popover more with styles, but it was quite constrained, I could not retain the exact width of the design.

File: client/profile-wizard/steps/store-details.js [hunk 4]
```diff
@@ -167,67 +177,110 @@ class StoreDetails extends Component {
...
-									onClick={ handleSubmit }
-									disabled={ ! isValidForm }
-								>
-									{ __( 'Continue', 'woocommerce-admin' ) }
-								</Button>
-							</Fragment>
-						) }
-					</Form>
-				</Card>
+							<CardFooter>
```
Comment (samueljseay) [inline client/profile-wizard/steps/store-details.js:232]: Making use of the new Card, CardFooter etc from `@wordpress/components`.

I've used 2 footers here to match the design

File: client/profile-wizard/steps/store-details.js [hunk 5]
```diff
@@ -167,67 +177,110 @@ class StoreDetails extends Component {
...
+												"I'm setting up a store for a client",
+												'woocommerce-admin'
+											) }
+											{ ...getInputProps( 'isClient' ) }
+										/>
+									</div>
+								</FlexItem>
+							</CardFooter>
+
+							<CardFooter justify="center">
```
Comment (samueljseay) [inline client/profile-wizard/steps/store-details.js:246]: The Card elements are flex components, so we can justify them or align them as desired to match the designs.

File: client/profile-wizard/style.scss [hunk 6]
```diff
@@ -72,9 +77,10 @@
 		margin-top: $gap-larger;
 		margin-left: auto;
 		margin-right: auto;
+		text-align: left;
 
 		> * {
-			max-width: 476px;
+			max-width: 504px;
```
Comment (samueljseay) [inline client/profile-wizard/style.scss:83]: The designs have made the card bigger to compensate for the larger amount of text content.

File: client/profile-wizard/style.scss [hunk 7]
```diff
@@ -93,9 +99,31 @@
 			font-weight: 400;
 		}
 
+		.components-popover__content {
```
Comment (samueljseay) [inline client/profile-wizard/style.scss:102]: Customizing the tooltip's popover element to get the desired display.

File: client/profile-wizard/style.scss [hunk 8]
```diff
@@ -93,9 +99,31 @@
...
+			> div {
+				padding: 11px 16px;
+			}
+		}
+
 		.woocommerce-card ~ p {
 			font-size: 14px;
 		}
+
+		.woocommerce-profile-wizard__footer {
```
Comment (samueljseay) [inline client/profile-wizard/style.scss:115]: a new footer element that holds the "skip wizard" link.

File: client/stylesheets/abstracts/_variables.scss [hunk 9]
```diff
@@ -6,6 +6,8 @@
 @import 'node_modules/@wordpress/base-styles/animations';
 @import 'node_modules/@wordpress/base-styles/z-index';
 
+@include wordpress-admin-schemes;
```
Comment (samueljseay) [inline client/stylesheets/abstracts/_variables.scss:9]: We need this to generate the css var that `$theme-color`is replaced by

File: client/stylesheets/abstracts/_variables.scss [hunk 10]
```diff
@@ -62,9 +64,9 @@ $wp-admin-background: #f1f1f1;
 $wp-admin-sidebar: #24292d;
 
 // Muriel
-$muriel-box-shadow-1dp: 0 2px 1px -1px rgba(0, 0, 0, 0.2), 0 1px 1px 0 rgba(0, 0, 0, 0.14),
-	0 1px 3px 0 rgba(0, 0, 0, 0.12);
-$muriel-box-shadow-6dp: 0 3px 5px rgba(0, 0, 0, 0.2), 0 1px 18px rgba(0, 0, 0, 0.12),
-	0 6px 10px rgba(0, 0, 0, 0.14);
-$muriel-box-shadow-8dp: 0 5px 5px -3px rgba(0, 0, 0, 0.2), 0 8px 10px 1px rgba(0, 0, 0, 0.14),
-	0 3px 14px 2px rgba(0, 0, 0, 0.12);
+$muriel-box-shadow-1dp: 0 2px 1px -1px rgba(0, 0, 0, 0.2),
```
Comment (samueljseay) [inline client/stylesheets/abstracts/_variables.scss:67]: My editor formatted these probably a prettier setting. Happy to revert this if its a pain.

File: packages/components/src/list/style.scss [hunk 11]
```diff
@@ -53,7 +54,7 @@
 	$background-color: $white;
 	$background-color-hover: $light-gray-100;
 	$border-color: $light-gray-tertiary;
-	$foreground-color: $theme-color;
+	$foreground-color: var(--wp-admin-theme-color);
```
Comment (samueljseay) [inline packages/components/src/list/style.scss:57]: use the new css var

File: packages/components/src/text-control/style.scss [hunk 12]
```diff
@@ -24,7 +22,22 @@
 		}
 	}
 
-	.components-text-control__input {
+	// @wordpress/components styles for text control target all types, so we must also do that
```
Comment (samueljseay) [inline packages/components/src/text-control/style.scss:25]: This is a change from the latest update to `@wordpress/components`

File: client/profile-wizard/steps/store-details.js (hunk 2 above)
Comment (psealock) [reply client/profile-wizard/steps/store-details.js:184]: No, its wrapped in `__()`, so the translations are handled elsewhere.

File: packages/components/src/list/style.scss (hunk 11 above)
Comment (psealock) [reply packages/components/src/list/style.scss:57]: `$theme-color` was derived from `theme(button)` in Gutenberg and evaluated in using postCSS. In our codebase we have variations like `theme(primary)` or `theme(outlines)`. Those may need to be replaced as well.

**Comment by samueljseay** [comment]
@psealock thanks for the feedback. I will separate out a PR for the base-styles update. and I'll look at the other theme variations. Strange I didn't see that CSS variable error in my environment, but I'll investigate.

About the translations, sorry my question wasn't very clear, I was more wondering, is there anything we need to do to provide translations for these new strings in other languages?

**Comment by samueljseay** [comment]
I'll close this for now and separate out a PR for the base-styles update first.

**Comment by psealock** [comment]
>About the translations, sorry my question wasn't very clear, I was more wondering, is there anything we need to do to provide translations for these new strings in other languages?

No we don't. Once on the .org repository, translators will translate the strings into different languages. See https://translate.wordpress.org/projects/wp-plugins/woocommerce/ for more on how thats done.

**Pull request closed without merging**